All tests are completed successfully

misha-ivanov opened the pull request:
test_bitfield, test_tset and sample_prime_numbers returns right result

misha-ivanov committed:
All tests are completed successfully

misha-ivanov committed:
Four gtests were added

Signed-off-by: Mike Ivanov <misha.ivanov.ig@gmail.com>

ArisovaAnastasia commented on src/tbitfield.cpp:
```diff
@@ -7,80 +7,193 @@
 int TBitField::GetMemIndex(const int n) const // индекс Мем для бита n
 {
+	if (n < 0)
+		throw "out_of_bottom_border!";
```
Здесь и ниже желательно в тексте исключения указывать название функции, где оно произошло

ArisovaAnastasia commented on src/tbitfield.cpp:
```diff
@@ -7,80 +7,193 @@
+		throw "out_of_bottom_border!";
+	if (n >= BitLen)
+		throw "out_of_upper_border!";
+	TELEM Mask = GetBit(n) << (sizeof(TELEM) * 8 - n % (sizeof(TELEM) * 8) - 1);
```
Подразумевается, что мы всегда сдвигаем именно 1, а не GetBit(n)

ArisovaAnastasia commented on src/tbitfield.cpp:
```diff
@@ -7,80 +7,193 @@
+	if (n >= BitLen)
+		throw "out_of_upper_border!";
+	TELEM Mask = 1;
+	Mask = Mask << (sizeof(TELEM) * 8 - (n % (sizeof(TELEM) * 8)) - 1);
```
Желательно воспользоваться уже написанной функцией выше GetMemMask

ArisovaAnastasia commented on src/tbitfield.cpp:
```diff
@@ -7,80 +7,193 @@
 
-TBitField TBitField::operator&(const TBitField &bf) // операция "и"
-{
+TBitField& TBitField::operator=(const TBitField& bf) // присваивание
```
Если this == &bf то дальнейший код можно не выполнять.
Также если размер MemLen у объектов совпадает, то память можно не перевыделять

ArisovaAnastasia commented on src/tbitfield.cpp:
```diff
@@ -7,80 +7,193 @@
+{
+	BitLen = bf.BitLen;
+	MemLen = bf.MemLen;
+	pMem = new TELEM[MemLen];
```
Прежде чем выделять новый блок памяти, надо удалить старый

ArisovaAnastasia commented on src/tset.cpp:
```diff
@@ -7,82 +7,138 @@
 // перегрузка ввода/вывода
 
-istream &operator>>(istream &istr, TSet &s) // ввод
+istream& operator>>(istream& istr, TSet& s) // ввод
```
При печати и вводе множества мы ожидаем увидеть все элементы из множества, а не набор 0 и 1

ArisovaAnastasia reviewed (commented):
Есть некоторые замечания, после их исправления лабораторная работа будет принята. Также надо удалить лишние файлы из репозитория и дописать свои тесты

misha-ivanov committed:
Some adjustments are added

Adjustments in methods: GetMemMask, SetBit, operator=, operator>> (TSet), operator<< (TSet)
Some comments about exceptions were corrected.

Signed-off-by: Mike Ivanov <misha.ivanov.ig@gmail.com>

ArisovaAnastasia commented:
Лабораторная работа принята
